neusomatic/python/extend_features.py

```diff
@@ -0,0 +1,378 @@
+
+    logger = logging.getLogger(extend_features.__name__)
+
+    logger.info(
+        "----------------------Extend Standalone Features------------------------")
+
+    if not os.path.exists(tumor_bam):
+        logger.error("Aborting!")
+        raise Exception("No tumor BAM file {}".format(tumor_bam))
+    if not os.path.exists(normal_bam):
+        logger.error("Aborting!")
+        raise Exception("No normal BAM file {}".format(normal_bam))
+    if not os.path.exists(tumor_bam + ".bai"):
+        logger.error("Aborting!")
+        raise Exception(
+            "No tumor .bai index file {}".format(tumor_bam + ".bai"))
+    if not os.path.exists(normal_bam + ".bai"):
+        logger.error("Aborting!")
+        raise Exception(
+            "No normal .bai index file {}".format(normal_bam + ".bai"))
+
+    if dbsnp:
+        if not os.path.exists(dbsnp):
+            logger.error("Aborting!")
+            raise Exception(
+                "No dbSNP file {}".format(dbsnp))
+
+        if dbsnp[-6:] != "vcf.gz":
+            logger.error("Aborting!")
+            raise Exception(
+                "The dbSNP file should be a tabix indexed file with .vcf.gz format")
+        if not os.path.exists(dbsnp + ".tbi"):
+            logger.error("Aborting!")
+            raise Exception(
+                "The dbSNP file should be a tabix indexed file with .vcf.gz format. No {}.tbi file exists.".format(dbsnp))
+
+    if cosmic:
+        cosmic_vars = {}
+        with open(cosmic) as i_f:
+            for line in i_f:
+                if not line.strip():
+                    continue
+                if line[0] == "#":
+                    continue
+                x = line.strip().split("\t")
+                chrom, pos, _, ref, alts, _, _, info = x[0:8]
+                num_cases = info.split("CNT=")[1].split(
+                    ";")[0] if "CNT=" in info else float('nan')
+                for alt in alts.split(","):
+                    var_id = "-".join([chrom, pos, ref, alt])
+                    cosmic_vars[var_id] = num_cases
+
+    if exclude_variants:
+        exclude_vars = []
+        with open(exclude_variants) as i_f:
+            for line in i_f:
+                if not line.strip():
+                    continue
+                if line[0] == "#":
+                    continue
+                if exclude_variants.split(".")[-1]=="tsv" and line[0:5]=="CHROM":
+                    continue
+                x = line.strip().split("\t")
+                chrom, pos, _, ref, alt = x[0:5]
+                var_id = "-".join([chrom, pos, ref, alt])
+                exclude_vars.append(var_id)
+
+    n_variants = 0
+    with open(candidates_vcf) as i_f:
+        for line in i_f:
+            if not line.strip():
+                continue
+            if line[0] == "#":
+                continue
+            n_variants += 1
+    logger.info("Number of variants: {}".format(n_variants))
+    split_len = n_variants // num_threads
```

Comment:
Shouldn't this be `(n_variants + num_threads - 1) // num_threads`, i.e., ceiling instead of floor for the division? Or will there be an extra thread with a small number of variants in some cases?